### PR TITLE
Data validation operators documentation

### PR DESCRIPTION
**Pull request by utkarsharma2**
# Description
## What is the current behavior?
Right now we don't have documentation for the data validation operator. 

related: #1607

## What is the new behavior?
Added documentation for data validation operator. 

## Does this introduce a breaking change?
Nope

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: python-sdk/example_dags/example_check_table.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+) as dag:
+    # [START data_validation__check_table]
+    imdb_movies = aql.load_file(
+        File("https://raw.githubusercontent.com/astronomer/astro-sdk/main/python-sdk/tests/data/homes.csv"),
+        output_table=Table(conn_id="sqlite_default"),
+    )
+    aql.check_table(
+        dataset=imdb_movies,
+        checks={
+            "sell_list": {"check_statement": "sell <= list"},
```
Comment (sunank200) [inline python-sdk/example_dags/example_check_table.py:23]: What does this mean here - `sell <=list`?

**Review by sunank200 — COMMENTED** [review]
Over all it looks good. But may be we should give more example of `check_column` and `check_table`. (may be with one realistic example DAG for a scenario) WDYT?

File: python-sdk/example_dags/example_check_table.py (hunk 1 above)
Comment (utkarsharma2) [reply python-sdk/example_dags/example_check_table.py:23]: Those a re column level checks and it's making sure all the values in `sell` col is less or equal to that `list` col

**Comment by sunank200** [comment]
> But may be we should give more example of `check_column` and `check_table`. (may be with one realistic example DAG for a scenario) WDYT?

@utkarsharma2

**Review by pankajastro — APPROVED** [review]
LGTM

**Pull request merged**